R/module_teal_data.R

```diff
@@ -53,14 +66,18 @@ srv_teal_data <- function(id,
                           data,
                           data_module,
                           modules = NULL,
-                          validate_shiny_silent_error = TRUE) {
+                          validate_shiny_silent_error = TRUE,
+                          failure_callback = function(data) {
+                            invisible(NULL)
+                          },
+                          is_transformer_failed = reactiveValues()) {
   checkmate::assert_string(id)
   checkmate::assert_class(data_module, "teal_data_module")
   checkmate::assert_multi_class(modules, c("teal_modules", "teal_module"), null.ok = TRUE)
```

Comment:
Should we add checks for `failure_callback` and `is_transformer_failed`?

```r
checkmate::assert_function(failure_callback)
checkmate::assert_class(is_transformer_failed, "reactivevalues")

```

Reply:
yes

Reply:
https://github.com/insightsengineering/teal/pull/1341/commits/a532670d911dff4a850740f27b02ce16ba437380